subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/AbstractMutableModuleComponentResolveMetadata.java

```diff
@@ -526,12 +526,14 @@ public boolean equals(Object o) {
                 && Objects.equal(excludes, that.excludes)
                 && Objects.equal(reason, that.reason)
                 && Objects.equal(attributes, that.attributes)
-                && Objects.equal(requestedCapabilities, that.requestedCapabilities);
+                && Objects.equal(requestedCapabilities, that.requestedCapabilities)
+                && Objects.equal(endorsing, that.endorsing)
+                && Objects.equal(dependencyArtifact, that.dependencyArtifact);
```

Comment:
❔ Is it concerning at all that changing the definition of equals here didn't break any tests?

Reply:
Honestly, I am not sure we use that contract much for that type. It could be a follow up investigation, but I believe we have more specific resolution issues to handle first.